File: Perfect arithmetic/Perfect arithmetic/countcarry.cpp
#include <iostream>
using namespace std;

int countCarry(int a, int b) {
    int carry = 0;
    int carryCount = 0;
    while (a > 0 || b > 0) {
        int digitA = a % 10;
        int digitB = b % 10;
        int sum = digitA + digitB + carry;
        carry = sum / 10;
        if (carry > 0) {
            carryCount++;
        }
        a /= 10;
        b /= 10;
    }
    return carryCount;
}

int main() {
    int n1, n2;
    cout << "Enter n1, n2: ";
    cin >> n1 >> n2;
    
    cout << countCarry(n1, n2) << " carry operations" << endl;
    return 0;
}

File: Perfect arithmetic/Perfect arithmetic/countcarry.py
def count_carry(a, b):
    carry = 0
    carry_count = 0
    while a > 0 or b > 0:
        digit_a = a % 10
        digit_b = b % 10
        carry, digit_sum = divmod(digit_a + digit_b + carry, 10)
        if carry > 0:
            carry_count += 1
        a //= 10
        b //= 10
    return carry_count

n1, n2 = map(int, input("Enter n1, n2: ").split())
print(count_carry(n1, n2), "carry operations.")
